docs/developer/addon-development.md

```diff
@@ -0,0 +1,64 @@
+---
+id: Add-on-development-guide
+title: "Add-on Development Guide"
+sidebar_position: 1
+sidebar_label: Add-on Development Guide
+keywords:
+  - Harvester
+  - harvester
+  - Add-on Development
+Description: How to write your own Harvester Add-on
+---
+
+# Background
+Harvester Add-ons provide a means to enable/disable specific Harvester or third party components based on the user requirements. 
+
+Add-ons provide a wrapper around [RKE2 HelmChart CRD](https://docs.rke2.io/helm#using-the-helm-crd).
+
+
+## Prerequisites
+
+- An existing Harvester cluster
+
+## add-on Spec
+
+The `Addon` CRD supports the following fields
+
+```yaml
+apiVersion: harvesterhci.io/v1beta1
+kind: Addon                         
+metadata:
+  name: example-add-on           # name of add-on
+  namespace: example-namespace  # namespace where add-on is deployed. The associated k8s components will be deployed in the same namespace as the addon crd
+  labels:                       # optional add-on labels
+    Add-on.harvesterhci.io/experimental: "true" # predefined label used by Harvester dashboard to indicate experimental tag in UI
+spec:
+  enabled: false                # boolean indicating if an add-on should be enabled or disabled on definition
+  repo: https://chartsrepo.com  # helm chart repo containing the helm chart being managed by add-on
+  version: "v0.19.0"            # version of helm chart to be installed
+  chart: mychart                # name of helm chart in the helm chart repo
+  valuesContent: |-             # values.yaml that needs to be passed to the helm chart
+    contents of values.yaml
+    that need to be passed
+    to the chart
+```
+
+An example add-on is the [rancher-vluster add-on.](https://github.com/harvester/experimental-addons/blob/main/rancher-vcluster/rancher-vcluster.yaml)
```

Comment:
```suggestion
:::note

Experimental add-ons are not directly packaged in Harvester. An example is the [rancher-vcluster](https://github.com/harvester/experimental-addons/blob/main/rancher-vcluster/rancher-vcluster.yaml) add-on.

:::
```

Reply:
I added a sentence to provide more context and connect the note to the preceding code block.